src/rtcModule/webrtcAdapter.cpp

```diff
@@ -85,20 +96,45 @@ rtc::scoped_refptr<webrtc::MediaStreamInterface> cloneMediaStream(
 
 void DeviceManager::enumInputDevices()
 {
-     if (!get()->GetVideoCaptureDevices(&(mInputDevices.video)))
-         throw std::runtime_error("Can't enumerate video devices");
-     get()->GetAudioInputDevices(&(mInputDevices.audio)); //normal to fail on iOS and other platforms that don't use device manager for audio devices
-//         throw std::runtime_error("Can't enumerate audio devices");
+    mInputDevices.audio.clear();
+    mInputDevices.video.clear();
+
+    std::unique_ptr<webrtc::VideoCaptureModule::DeviceInfo> info(
+        webrtc::VideoCaptureFactory::CreateDeviceInfo());
+    if (!info)
+    {
+#ifndef __ANDROID__
+        throw std::runtime_error("Can't enumerate video devices");
+#else
+        return;
+#endif
+    }
+
+    int numDevices = info->NumberOfDevices();
+    auto& devices = mInputDevices.video;
+    for (int i = 0; i < numDevices; ++i)
+    {
+        const uint32_t kSize = 256;
+        char name[kSize] = {0};
+        char id[kSize] = {0};
+        if (info->GetDeviceName(i, name, kSize, id, kSize) != -1)
+        {
+            devices.push_back(cricket::Device(name, id));
+        }
+    }
+    // TODO: Implement audio device enumeration, when webrtc has it again
+    // Maybe VoEHardware in src/voice_engine/main/interface/voe_hardware.h
+     mInputDevices.audio.push_back(cricket::Device("default", "0"));
```

Comment:
Move this line upwards, to the beginning of the method, so in case of Android, we have at least the default device.